source/tutorials/batch-inference/csv-batch-inference.rst

```diff
@@ -0,0 +1,160 @@
+
+    We're also going to need two files:
+
+    * a model trained with TensorFlow 2.5.0
+    * some new data in a single CSV file
+
+    To make it easy for you they are available here, no need to download them:
+
+    * Model: `<s3://valohai-public-files/tutorials/batch-inference/csv-batch-inference/model.zip>`_
+    * Images: `<s3://valohai-public-files/tutorials/batch-inference/csv-batch-inference/data.csv>`_
+
+    If you want to, you can train the required model by following the Keras example here: `<https://keras.io/examples/structured_data/structured_data_classification_from_scratch/>`_.
+
+Running on Valohai
+------------------
+
+To easily run our batch inference on Valohai, we will use it to run our code from the very beginning.
+
+If you don't already have a Valohai account, go to `<https://app.valohai.com/>`_ to create one for yourself.
+
+Create a new folder for our project, then run the following commands in the project folder:
+
+.. code-block:: bash
+
+    vh login
+    # fill in your username
+    # and your password
+
+    vh init
+    # Answer the wizard questions like this:
+    # "First, let's..." -> y
+    # "Looks like..." -> python batch_inference.py, then y to confirm
+    # "Choose a number or..." -> tensorflow/tensorflow:2.5.0, then y to confirm
+    # "Write this to..." -> y
+    # "Do you want to link..." -> C, then give a name for your project, then select your user
+
+Edit the generated ``valohai.yaml`` so that it looks like this:
+
+.. code-block:: yaml
+
+    ---
+
+    - step:
+        name: Batch Inference
+        image: tensorflow/tensorflow:2.5.0
+        command:
+        - pip install pandas valohai-utils
+        - python batch_inference.py
+        inputs:
+        - name: model
+          default: s3://valohai-public-files/tutorials/batch-inference/csv-batch-inference/model.zip
+        - name: images
+          default: s3://valohai-public-files/tutorials/batch-inference/csv-batch-inference/data.csv
+
+What we are doing here is defining a single step for our machine learning pipeline, which is the Batch Inference step. We will run on top of the official ``tensorflow/tensorflow:2.5.0`` Docker image, first install the ``valohai-utils`` Python library and then run our batch inference code.
+
+Let's test that everything is set up correctly by running on Valohai:
+
+.. code-block:: bash
+
+    vh exec run --adhoc "Batch Inference"
+
+If everything went as planned, we should see our Valohai execution end after finding out that ``batch_inference.py`` is missing:
+
+.. image:: batch-inference-tutorial-1.png
+   :alt: Error, but success!
+
+Unpacking the Model
+--------------------
+
+Today we are unpacking the model ourselves. Let's get started by creating and opening up ``batch_inference.py`` in your favorite editor!
+
+Add these imports to the beginning of the file:
+
+.. code-block:: python
+
+    import json
+    from zipfile import ZipFile
+
+    import pandas as pd
```

Comment:
I really think using Pandas for a simple CSV file is overkill (the `csv` module is built-in), but I also kinda get the point of showing the use of tools data scientists might be familiar with.

Installing Pandas (and its dependencies) balloons a freshly created Python virtualenv from 8 megs to 157 megs.

Reply:
Yeah. I go with Pandas as that is what is used in the model training tutorial.